### PR TITLE
Documenting enablement of UART1 support for nrf52840_pca10056

### PR DESCRIPTION
**Pull request by laperie**
Adding a set of BKMs on how to enable and configure UART1
for the nrf52840_pca10056 board. This instructions are likely valid
to most other nrf52840- family of boards.

Signed-off-by: Andrei Laperie <andrei.laperie@intel.com>

### CODE REVIEW

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: boards/arm/nrf52840_pca10056/doc/nrf52840_pca10056.rst [hunk 1]
```diff
@@ -167,6 +167,54 @@ You can build and flash the examples to make sure Zephyr is running correctly on
...
 
+Using UART1
+***********
+
+The following approach can be used when an application needs to use
+more than one UART for connecting peripheral devices:
+
+1. Add device tree overlay file to the main directory of your application
+
+.. code-block:: console
```
Comment (dbkinder) [inline boards/arm/nrf52840_pca10056/doc/nrf52840_pca10056.rst]: For this code block to be part of the bullet list, it needs to be indented, as does the sentence after the code block.  Try this:

```
1. Add device tree overlay file to the main directory of your application:

   .. code-block:: console

      $ cat nrf52840_pca10056.overlay
      &uart1 {
        compatible = "nordic,nrf-uarte";
        current-speed = <115200>;
        status = "ok";
        tx-pin = <14>;
        rx-pin = <16>;
      };

   In the overlay file above, pin P0.16 is used for RX and P0.14 is used for TX
```

File: boards/arm/nrf52840_pca10056/doc/nrf52840_pca10056.rst [hunk 2]
```diff
@@ -167,6 +167,54 @@ You can build and flash the examples to make sure Zephyr is running correctly on
...
+    compatible = "nordic,nrf-uarte";
+    current-speed = <115200>;
+    status = "ok";
+    tx-pin = <14>;
+    rx-pin = <16>;
+  };
+
+In the overlay file above, pin P0.16 is used for RX and P0.14 is used for TX
+
+2. Go to menuconfig and enable :option: `UART_1_NRF_UARTE`:
```
Comment (dbkinder) [inline boards/arm/nrf52840_pca10056/doc/nrf52840_pca10056.rst]: No space after the ``:option:`` and we need to start the ``(top menu)...`` line as a new paragraph:
```
2. Go to menuconfig and enable :option:`UART_1_NRF_UARTE`:

   (top menu) -> Device Drivers -> Serial Drivers -> nRF UART nrfx drivers
```

File: boards/arm/nrf52840_pca10056/doc/nrf52840_pca10056.rst [hunk 3]
```diff
@@ -167,6 +167,54 @@ You can build and flash the examples to make sure Zephyr is running correctly on
...
+   (top menu) -> Device Drivers -> Serial Drivers -> nRF UART nrfx drivers
+
+3. Use the UART1 as device_get_binding("UART_1")
```
Comment (dbkinder) [inline boards/arm/nrf52840_pca10056/doc/nrf52840_pca10056.rst]:
```suggestion
3. Use the UART1 as ``device_get_binding("UART_1")``
```

File: boards/arm/nrf52840_pca10056/doc/nrf52840_pca10056.rst [hunk 4]
```diff
@@ -167,6 +167,54 @@ You can build and flash the examples to make sure Zephyr is running correctly on
...
+Overlay file naming
+===================
+The file has to be named ``<board>.overlay`` and placed in the app main directory to be
+picked up automatically by the device tree compiler.
+
+Selecting the pins
+==================
+To select the pin numbers for tx-pin and rx-pin:
+
+.. code-block::.. console
```
Comment (dbkinder) [inline boards/arm/nrf52840_pca10056/doc/nrf52840_pca10056.rst]: Change to:
```
.. code-block:: console

   tx-pin = <pin_no>
```

File: boards/arm/nrf52840_pca10056/doc/nrf52840_pca10056.rst [hunk 5]
```diff
@@ -167,6 +167,54 @@ You can build and flash the examples to make sure Zephyr is running correctly on
...
+	tx-pin = <pin_no>
+
+Open the `nRF52840 Product Specification`_, chapter 7 'Hardware and Layout'.
+In the table 7.1.1 'aQFN73 ball assignments' select the pins marked
+'General purpose I/O'.  Note that pins marked as 'low frequency I/O only' can only be used
+in under-10KHz applications. They are not sutiable for 115200 speed of UART.
+
+Translating 'Pin' into number for Device tree using following formula:
```
Comment (dbkinder) [inline boards/arm/nrf52840_pca10056/doc/nrf52840_pca10056.rst]: Change to (note the ``::`` at the end of this first line, and the indent and blank lines):
```
Translate 'Pin' into number for Device tree by using the following formula::

   pin_no = b\*32 + a

where ``a`` and ``b`` are from the Pin value in the table (Pb.a).
For example, for P0.1, ``pin_no = 1`` and for P1.0, ``pin_no = 32``.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
Almost there on the doc changes

**Review by jakub-uC — APPROVED** [review]

**Comment by jarz-nordic** [comment]
@carlescufi : FYI

**Review by carlescufi — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
LGTM, thanks!

**Pull request merged**